### DISCUSSION TITLE
How do I debug fast-api server in intelliJ

### DISCUSSION THREAD

**Discussion by BainDragneel** [fastapi/fastapi#7214] [Questions] [answered]
### Environment

* OS:  macOS 
* FastAPI Version : 0.61.1

* Python version: 3.9.0
* intelliJ version: 2020.2.3 , #IU-202.7660.26
* python plugin version:  202.7660.27
```
import uvicorn
from app import create_app
from app.route import router

app = create_app()
app.include_router(router, prefix="/api")

if __name__ == "__main__":
    uvicorn.run(app='main:app', host="127.0.0.1", port=6666, debug=True)
```
I can use uvicorn command line to debug, url can be requested. But using the intellij debug main.py file, the output:
```
pydev debugger: process 67720 is connecting
Connected to pydev debugger (build 202.7660.26)
INFO:     Uvicorn running on http://127.0.0.1:6666 (Press CTRL+C to quit)
INFO:     Started reloader process [67720] using statreload
INFO:     Started server process [67722]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
```
But the url cannot be requested!

**Comment by Mause** [comment]
What happens when you try to call the server?

**Comment by BainDragneel** [comment]
use intelliJ debug:
![image](https://user-images.githubusercontent.com/47975259/98354260-67c4de00-205b-11eb-99e4-8a707601941c.png)
![image](https://user-images.githubusercontent.com/47975259/98354345-888d3380-205b-11eb-845c-1115328e0351.png)
![image](https://user-images.githubusercontent.com/47975259/98354376-95118c00-205b-11eb-9424-f3748b7e2e56.png)
 use uvicorn command line with default port to debug:
![image](https://user-images.githubusercontent.com/47975259/98354684-06e9d580-205c-11eb-8e59-804efdaffdbf.png)
![image](https://user-images.githubusercontent.com/47975259/98354591-e9b50700-205b-11eb-8c97-972d60f9c2d9.png)

and i see  use uvicorn command line with the port 6666  to debug: the url cannot be requested!

**Comment by BainDragneel** [comment]
Seems to be a port problem!

**Comment by BainDragneel** [comment, marked as answer]
I will close this issue, because i find  that just modify the port! So sorry! 
I will check my 6666 port

**Comment by tiangolo** [comment]
Thanks for the help @Mause ☕ 

And thanks for coming back to close the issue @BainDragneel 🍰 

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
